server/server.js

```diff
@@ -0,0 +1,115 @@
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts";
+mongoose
+  .connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+  .catch((error) => {
+    console.log(`Cannot connect to database: ${error}`);
+  });
+mongoose.Promise = Promise;
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+app.use(cors());
+app.use(bodyParser.json());
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res
+      .status(500)
+      .json({ error: "Service unavailable" });
+  }
+});
+
+const Thought = new mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: [true, "Thought is required"],
+    validate: [
+      {
+        validator: (thoughtValue) => {
+          return thoughtValue.length >= 5;
+        },
+        message: "Thought must be longer than 5 characters, please try again.",
+      },
+      {
+        validator: (thoughtValue) => {
+          return thoughtValue.length < 140;
+        },
+        message:
+          "Thought must be shorter than 140 characters, please try again",
+      },
+    ],
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+});
+
+if (process.env.RESET_DATABASE) {
+  const resetDatabase = async () => {
+    await Thought.deleteMany();
+  };
+  resetDatabase();
+}
+const listEndpoints = require('express-list-endpoints');
```

Comment:
You shouldn't mix  2 methods of importing external modules. Choose between `import` and `require` syntax and be consistent. 😸